packages/calcite-components/src/components/input-date-picker/input-date-picker.tsx

```diff
@@ -1103,11 +1115,19 @@ export class InputDatePicker
                   }}
                   disabled={disabled}
                   icon="calendar"
-                  label={this.label}
+                  label={
+                    this.range
+                      ? this.startLabel || this.messages.startInput
```

Comment:
I think for all these we should be using the null coalescing operator `??` instead of `||` so that someone could set the label to an empty string if necessary. Applies to all other areas